libres/lib/enkf/enkf_main_manage_fs.cpp

```diff
@@ -128,60 +128,40 @@ void enkf_main_initialize_from_scratch(
     }
 }
 
-static void enkf_main_copy_ensemble(
-    const enkf_main_type *enkf_main, enkf_fs_type *source_case_fs,
-    int source_report_step, enkf_fs_type *target_case_fs,
-    int target_report_step, const std::vector<bool> &iens_mask,
-    const char *
-        ranking_key, /* It is OK to supply NULL - but if != NULL it must exist */
-    const std::vector<std::string> &node_list) {
-    const int ens_size = enkf_main_get_ensemble_size(enkf_main);
+static void enkf_main_copy_ensemble(const ensemble_config_type *ensemble_config,
+                                    enkf_fs_type *source_case_fs,
+                                    int source_report_step,
+                                    enkf_fs_type *target_case_fs,
+                                    const std::vector<bool> &iens_mask,
+                                    const std::vector<std::string> &node_list) {
     state_map_type *target_state_map = enkf_fs_get_state_map(target_case_fs);
 
     {
-        int *ranking_permutation;
-        int inode, src_iens;
-
-        if (ranking_key != NULL) {
-            ranking_table_type *ranking_table =
-                enkf_main_get_ranking_table(enkf_main);
-            ranking_permutation = (int *)ranking_table_get_permutation(
-                ranking_table, ranking_key);
-        } else {
-            ranking_permutation =
-                (int *)util_calloc(ens_size, sizeof *ranking_permutation);
-            for (src_iens = 0; src_iens < ens_size; src_iens++)
-                ranking_permutation[src_iens] = src_iens;
-        }
 
         for (auto &node : node_list) {
-            enkf_config_node_type *config_node = ensemble_config_get_node(
-                enkf_main_get_ensemble_config(enkf_main), node.c_str());
-            for (src_iens = 0;
-                 src_iens < enkf_main_get_ensemble_size(enkf_main);
-                 src_iens++) {
-                if (src_iens < iens_mask.size() && iens_mask[src_iens]) {
-                    int target_iens = ranking_permutation[src_iens];
+            enkf_config_node_type *config_node =
+                ensemble_config_get_node(ensemble_config, node.c_str());
+
+            int src_iens = 0;
+            for (auto mask : iens_mask) {
+                if (mask) {
                     node_id_type src_id = {.report_step = source_report_step,
```

Comment:
source_report_step is also 0 I guess

Reply:
One would think so, and it probably is, but it is possible to provide by the user, so would be a breaking change.